### ISSUE TITLE
Create tests to be run in the Coq implementation

### ISSUE THREAD

**Issue by jldodds** [cjerdonek/formal-rcv#10]
Before closing #2 I'd like to run a few tests. Any ideas for interesting situations to test are welcome.

**Comment by cjerdonek** [comment]
Hi @jldodds, my approach is usually to go simultaneously "from the inside out" and "from the outside in."

So I'd start with--
1. Some unit tests in Coq of some of the innermost functions (preferably the dicier ones if there are any), and to cover all the edge cases of those functions to have a solid foundation to build on.  This also helps to ensure that you've decomposed your implementation in a good way.
2. At least one "end-to-end" test of a non-trivial election, which should get you most of the code paths.  For example, I'd start with one with at least two rounds.  My advice would be to go the route of issue #7 by extracting the implementation and specifying the input data as JSON (e.g. like what I started working on here: https://github.com/cjerdonek/open-rcv-tests ).  After that, you could add JSON data for (1) an election having a tie for last place in some round, and (2) an election with batch elimination in some round.  It would be straightforward to add more once you have this infrastructure in place.

**Comment by jldodds** [comment]
Is there any actual election data available to use as a test case?

**Comment by cjerdonek** [comment]
The SF Department of Elections has the ballot data for all past SF elections, but it's in a format that's a little annoying to use: http://www.sfelections.org/

An alternative would be to grab data from here (click on the link that says, "OpenSTV's ballot data repository"): http://www.openstv.org/node/258

That page has data in BLT format which is easier to parse.

**Comment by jldodds** [comment]
Thanks, are there any elections that jump to mind as particularly interesting?

**Comment by cjerdonek** [comment]
Yes, probably the _most_ interesting (in the sense of most rounds, candidates, etc) is the District 10 Supervisor race from November 2, 2010.  Malia Cohen was the winner of that race: http://sfelections.org/results/20101102/data/d10.html
